benchmark-overhead/README.md

```diff
@@ -1,27 +1,107 @@
 
 # Overhead tests
 
-This is a work in progress. 
+* [Process](#process)
+* [What do we measure?](#what-do-we-measure)
+* [Config](#config)
+  + [Agents](#agents)
+* [Automation](#automation)
+* [Setup and Usage](#setup-and-usage)
+* [Visualization](#visualization)
 
 This directory will contain tools and utilities
 that help us to measure the performance overhead introduced by 
 the agent and to measure how this overhead changes over time.
 
-## Contents
+The overhead tests here should be considered a "macro" benchmark. They serve to measure high-level
+overhead as perceived by the operator of a "typical" application. Tests are performed on an JDK 11 
+virtual machine from [AdoptOpenJDK](https://adoptopenjdk.net/).
 
-tbd
+## Process
+
+There is one dynamic test here called [OverheadTests](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/benchmark-overhead/src/test/java/io/opentelemetry/OverheadTests.java). 
+The `@TestFactory` method creates a test pass for each of the [defined configurations](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/benchmark-overhead/src/test/java/io/opentelemetry/config/Configs.java).
+Before the tests run, a single collector instance is started. Each test pass has one or more agents configured and those are tested in series.
+For each agent defined in a configuration, the test runner (using [testcontainers](https://www.testcontainers.org/)) will:
+1. create a fresh postgres instance and populate it with initial data.
+2. create a fresh instance of [spring-petclinic-rest](https://github.com/spring-petclinic/spring-petclinic-rest) instrumented with the specified agent
+3. measure the time until the petclinic app is marked "healthy" and then write it to a file
+4. if configured, perform a warmup phase. During the warmup phase, a bit of traffic is generated in order to get the application into a steady state (primarily helping facilitate jit compilations). Currently, we use a 30 second warmup time. 
+5. start a JFR recording by running `jcmd` inside the petclinic container
+6. run the [k6 test script](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/benchmark-overhead/k6/basic.js) with the configured number of iterations through the file and the configured number of concurrent virtual users (VUs).
+7. after k6 completes, petclinic is shut down
+8. after petclinic is shut down, postgres is shut down
+
+And this repeats for every agent configured in each test configuration.
+
+After all the tests are complete, the results are collected and commited back to the `/results` subdirectory as csv and summary text files.
+
+## What do we measure?
+
+For each test pass, we record the following metrics in order to compare agents and determine
+relative overhead.
+
+| metric name              | description                                                     |
```

Comment:
Please add units to all rows

Reply:
And probably it will be useful to have them in summary reports as well

Reply:
Ok, added a units column. Let's do the report change in a separate PR?  Mind filing an issue?

Reply:
filed #3950